src/main/java/seedu/sudowudo/model/item/Item.java

```diff
@@ -57,200 +53,197 @@ public Item(Description desc, UniqueTagList tags) {
-		case "overdue":
-			return this.is("task") && this.getEndDate() != null 
-			        && this.getEndDate().isBefore(LocalDateTime.now());
-		case "item":
-		    return true;
-		default:
-			return false;
-		}
+    public String getType() {
+        if (this.getStartDate() != null) {
+            return "Event";
+        } else if (this.getEndDate() == null) {
+            return "Floating Task";
+        } else {
+            return "Task";
+        }
     }
 
-	/**
-	 * Returns the type of the item. Useful for Item cards and messages.
-	 * 
-	 * @return The type of the item, computed based on start/end dates.
-	 *         [Event|Floating Task|Task]
-	 * @@author A0092390E
-	 */
-	@Override
-	public String getType() {
-		if (this.getStartDate() != null) {
-			return "Event";
-		} else if (this.getEndDate() == null) {
-			return "Floating Task";
-		} else {
-			return "Task";
-		}
-	}
-
-	/**
-	 * Replaces this Item's tags with the tags in the argument tag list.
-	 */
+    /**
+     * Replaces this Item's tags with the tags in the argument tag list.
+     */
     public void setTags(UniqueTagList replacement) {
         tags.setTags(replacement);
-		setChanged();
-		notifyObservers();
+        setChanged();
+        notifyObservers();
     }
 
-    
     @Override
-	public void setIsDone(boolean doneness) {
+    public void setIsDone(boolean doneness) {
         this.isDone = doneness;
-		setChanged();
-		notifyObservers();
+        setChanged();
+        notifyObservers();
     }
 
-    public void setStartDate(LocalDateTime startDate) {
+    public void setStartDate(LocalDateTime startDate) throws IllegalValueException {
+        if (!isValidInterval(startDate, this.endDate)) {
+            throw new IllegalValueException(MESSAGE_DATE_CONSTRAINTS);
+        }
         this.startDate = startDate;
-		setChanged();
-		notifyObservers();
+        setChanged();
+        notifyObservers();
     }
 
-    public void setEndDate(LocalDateTime endDate) {
+    public void setEndDate(LocalDateTime endDate) throws IllegalValueException {
+        if (!isValidInterval(this.startDate, endDate)) {
+            throw new IllegalValueException(MESSAGE_DATE_CONSTRAINTS);
+        }
+        this.endDate = endDate;
+        setChanged();
+        notifyObservers();
+    }
+
+    public void setPeriod(LocalDateTime startDate, LocalDateTime endDate) throws IllegalValueException {
+        if (!isValidInterval(startDate, endDate)) {
+            throw new IllegalValueException(MESSAGE_DATE_CONSTRAINTS);
+        }
+        this.startDate = startDate;
         this.endDate = endDate;
-		setChanged();
-		notifyObservers();
+        setChanged();
+        notifyObservers();
     }
 
-    public void setRecurring(boolean isRecurring) {
-        this.isRecurring = isRecurring;
-		setChanged();
-		notifyObservers();
+    private boolean isValidInterval(LocalDateTime start, LocalDateTime end) {
+        return start == null || end == null || start.isBefore(end);
```

Comment:
Doesn't handle `Item`s with null `startDate` or `endDate` properly. A task has a null `startDate`. A floating task has null `startDate` and `endDate`.